sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/Samplers.java

```diff
@@ -147,14 +147,27 @@ public static Sampler alwaysOff() {
    * whether or not to sample. If there is no parent, the Sampler uses the provided Sampler delegate
    * to determine the sampling decision.
    *
-   * @param delegateSampler the {@code Sampler} which is used to make the sampling decisions if the
-   *     parent does not exist.
+   * @param root the {@code Sampler} which is used to make the sampling decisions if the parent does
+   *     not exist.
    * @return a {@code Sampler} that follows the parent's sampling decision if one exists, otherwise
-   *     following the delegate sampler's decision.
+   *     following the root sampler's decision.
    * @since 0.7.0
    */
-  public static Sampler parentOrElse(Sampler delegateSampler) {
-    return new ParentOrElse(delegateSampler);
+  public static Sampler parentBased(Sampler root) {
+    return newParentBasedBuilder(root).build();
+  }
+
+  /**
+   * Returns a {@link ParentBased.Builder} that follows the parent's sampling decision if one
+   * exists, otherwise following the root sampler and other optional sampler's decision.
+   *
+   * @param root the required {@code Sampler} which is used to make the sampling decisions if the
+   *     parent does not exist.
+   * @return a {@code ParentBased.Builder}
+   * @since 0.8.0
+   */
+  public static ParentBased.Builder newParentBasedBuilder(Sampler root) {
```

Comment:
Also to @jkwatson I think we should either have `new` on both or `new` on neither the non-builder and builder factories, they seem equally new. wdyt?

Reply:
Usually, If we provide `builder ` factories, we should keep the `constructor ` private. right?

Reply:
@anuraaga you were suggesting that the `parentBased(Sampler)` method should be `newParentBased(Sampler)`, if we want them to match, correct?

Reply:
Ah yeah I was just talking about the factory names, not about changing constructor visibility. Indeed `newParentBased(Sampler)`. Alternatively, rename the builder returning one to `parentBasedBuilder`

Reply:
@anuraaga Gotcha 👍 thanks. what about this?
```
parentBased(Sampler root)
parentBasedBuilder(Sampler root)
```

Reply:
I'm good with that :)